### DISCUSSION TITLE
Solving duplicate logging with podman 3.x and systemd.

### DISCUSSION THREAD

**Discussion by markstos** [containers/podman#20677] [Q&A] [answered]
I'm running Podman 3 on a Ubuntu server that doesn't have Podman 4 packages available. We are getting duplicate logging out of the containers in the systemd journal.

I've researched the issue, and it appears `--log-driver=passthrough` was added as a solution to this. 

I have Podman 4 on my laptop, so I was curious of Quadlet would set up particular details for logging, like adding `--log-driver=passthrough`. It does not. 

But from searching through issues here, it appears that --log-driver=passthrough /was/ added as the default for Quadlet, and then later reverted because it doesn't work for rootless containers (#17692). 

So I just want to check my understanding that the way to solve duplicate logging for rootful containers under systemd is to update my OS to a version that supports Podman 4 (or switch to Red Hat!) and then use the `--log-driver=passthrough` option. 

Thanks!

**Comment by ygalblum** [comment, marked as answer]
You are correct, initially Quadlet used to set the log driver without allowing the user to change it. This was due to the fact that initially it was meant to be more opinionated. However, as it evolved, it was decided to remove most of its opinions and keep podman's defaults along with allowing the user to override them.
As a result, in order to use a non-default log driver (e.g. `passthrough`) users need to set `LogDriver=passthrough` in their Quadlet file.
